service/src/test/java/uk/gov/hmcts/reform/fpl/service/cmo/HearingOrderKindEventDataBuilderTest.java

```diff
@@ -0,0 +1,296 @@
+import uk.gov.hmcts.reform.fpl.model.event.UploadDraftOrdersData;
+import uk.gov.hmcts.reform.fpl.model.order.HearingOrder;
+import uk.gov.hmcts.reform.fpl.model.order.HearingOrdersBundle;
+import uk.gov.hmcts.reform.fpl.service.IdentityService;
+
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.element;
+
+class HearingOrderKindEventDataBuilderTest {
+
+    private static final UUID SELECTED_HEARING_ID = UUID.randomUUID();
+    private static final DocumentReference DOCUMENT_REFERENCE_0 = mock(DocumentReference.class);
+    private static final DocumentReference DOCUMENT_REFERENCE_1 = mock(DocumentReference.class);
+    private static final DocumentReference DOCUMENT_REFERENCE_2 = mock(DocumentReference.class);
+    private static final DocumentReference DOCUMENT_REFERENCE_3 = mock(DocumentReference.class);
+    private static final DocumentReference DOCUMENT_REFERENCE_4 = mock(DocumentReference.class);
+    private static final DocumentReference DOCUMENT_REFERENCE_5 = mock(DocumentReference.class);
+    private static final DocumentReference DOCUMENT_REFERENCE_6 = mock(DocumentReference.class);
+    private static final DocumentReference DOCUMENT_REFERENCE_7 = mock(DocumentReference.class);
+    private static final DocumentReference DOCUMENT_REFERENCE_8 = mock(DocumentReference.class);
+    private static final DocumentReference DOCUMENT_REFERENCE_9 = mock(DocumentReference.class);
+    private static final UUID UUID_1 = UUID.randomUUID();
+
+    private final IdentityService identityService = mock(IdentityService.class);
+
+    private final HearingOrderKindEventDataBuilder underTest = new HearingOrderKindEventDataBuilder(identityService,
+        new OptionCountBuilder());
+
+    @Test
+    void testEmptyHearingOrderBundleDraft() {
+        when(identityService.generateId()).thenReturn(UUID_1);
+
+        UploadDraftOrdersData.UploadDraftOrdersDataBuilder newEventDataBuilder = UploadDraftOrdersData.builder();
+
+        underTest.build(SELECTED_HEARING_ID, CaseData.builder().build(), UploadDraftOrdersData.builder().build(),
+            newEventDataBuilder);
+
+        assertThat(newEventDataBuilder.build()).isEqualTo(UploadDraftOrdersData.builder()
+            .currentHearingOrderDrafts(List.of(element(UUID_1, HearingOrder.builder().build())))
+            .orderToSendOptionCount("0")
+            .orderToSend0(null)
+            .orderToSend1(null)
+            .orderToSend2(null)
+            .orderToSend3(null)
+            .orderToSend4(null)
+            .orderToSend5(null)
+            .orderToSend6(null)
+            .orderToSend7(null)
+            .orderToSend8(null)
+            .orderToSend9(null)
+            .build()
+        );
+    }
+
+    @Test
+    void testExistingHearingOrderBundleDraftWitC21() {
```

Comment:
Minor typo, should have "testExistingHearingOrderBundleDraftWitC21" as "testExistingHearingOrderBundleDraftWithC21"

Reply:
👍